packages/nextjs/src/middleware/utils/injectAuthIntoRequest.ts

```diff
@@ -0,0 +1,15 @@
+import { GetServerSidePropsContext } from 'next';
+
+import { AuthData, ContextWithAuth } from '../types';
+
+/**
+ * Inject `auth`, `user` and `session` properties into ctx.request
+ * @internal
+ */
+export function injectAuthIntoRequest(ctx: GetServerSidePropsContext, authData: AuthData): ContextWithAuth {
+  const { user, session, userId, sessionId, getToken } = authData;
+  (ctx.req as any).auth = { userId, sessionId, getToken };
```

Comment:
@igneel64 @SokratisVidros 
Instead of making a copy of `ctx` and `req`, I decided to mutate the original and inject the auth state directly - both are class instances and not plain objects so this should be safer. unless I'm wrong, most express middlewares mutate the original req directly so I'm following the same practise here.  Can you think of any scenarios where this could be considered problematic?

Reply:
Not at all, only typings but I don't see any danger on that now.